website/components/AppWrapper.tsx

```diff
@@ -110,17 +118,19 @@ export default function AppWrapper(inputProps: AppWrapperProps): ReactElement {
         <FlexRowAlignCenter $gap={15}>
           <FlexRowAlignCenter $gap={2}>
             <LoadModal onLoad={onLoad} />
-            <ShareModal appProps={viewerArgs} />
+            {viewerArgs && <ShareModal appProps={viewerArgs} />}
           </FlexRowAlignCenter>
           {/* <HelpDropdown /> */}
         </FlexRowAlignCenter>
       </Header>
-      <ImageViewerApp
-        {...viewerArgs}
-        appHeight={`calc(100vh - ${HEADER_HEIGHT_PX}px)`}
-        canvasMargin="0 0 0 0"
-        viewerSettings={viewerSettings}
-      />
+      {viewerArgs && viewerSettings && (
+        <ImageViewerApp
+          {...viewerArgs}
+          appHeight={`calc(100vh - ${HEADER_HEIGHT_PX}px)`}
+          canvasMargin="0 0 0 0"
+          viewerSettings={viewerSettings}
+        />
+      )}
```

Comment:
Conditional rendering, waits until the args have been initialized.

Reply:
I understand you made this change based on "It turns out that the ImageViewerApp only uses passed props as initial values, and ignores all future updates to them. " but that seems like a weird behavior in a react app? 

Also to make the meaning clearer, I might just make a flag called `processedUrlParamsComplete` or something instead of this double exists

Reply:
That makes sense, yeah! And yes, I agree the behavior is weird. There's an issue ticket here for a potential refactor to move state handling out of the main ImageViewerApp: #235